src/sql/workbench/contrib/notebook/browser/models/notebookInput.ts

```diff
@@ -319,6 +320,11 @@ export abstract class NotebookInput extends EditorInput implements INotebookInpu
 		return this._connectionProfile;
 	}
 
+	public setNotebookContents(value: azdata.nb.INotebookContents) {
```

Comment:
I'd suggest just making this a property with only a setter. That way we can easily add a getter later on if we need without needing to have two separate functions.

Reply:
If we have a setter, then we have to have a getter. You can still call a getter if it doesn't exist, it just returns undefined.

Reply:
We pass this around as an INotebookInput interface, so we're limited by the access modifiers there. There's a readonly, but no writeonly.

Reply:
*shrug* Doesn't seem like a problem to me. (and why not just add the property now to get the contents? Save having to refactor this later if we want to). 

But guess it's not a huge deal. If you feel strongly about the current functionality then alrighty.

Reply:
Sure, I can do that. Although I should mention that I removed the getter earlier because you asked: https://github.com/microsoft/azuredatastudio/pull/18972#discussion_r846276985

Reply:
hah, I'm nothing if not inconsistent. Well clearly I have no strong reasoning either way then so whatever you think is best, although I guess for an internal call like this having the setter function like this is fine.